### PR TITLE
Update README: HaveIBeenPwned v2 to v3

### PR DESCRIPTION
**Pull request by VorBoto**
Thank you for taking the time to work on a Pull Request for this project!

To ensure your PR is dealt with swiftly please check the following:

- [ ] Your submissions are formatted according to the guidelines in the [contributing guide](CONTRIBUTING.md)
- [ ] Your additions are ordered alphabetically
- [ ] Your submission has a useful description
- [ ] The description does not end with punctuation
- [ ] Each table column should be padded with one space on either side
- [ ] You have searched the repository for any relevant issues or pull requests
- [ ] Any category you are creating has the minimum requirement of 3 items
- [ ] All changes have been [squashed][squash-link] into a single commit

[squash-link]: <https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit>

### CODE REVIEW

**Comment by Jamim** [comment]
Corresponding issue: #1009

File: README.md [hunk 1]
```diff
@@ -662,7 +662,7 @@ API | Description | Auth | HTTPS | CORS |
 |---|---|---|---|---|
 | [AXFR Database](http://api.axfrcheck.com) | AXFR public database | No | No | Unknown |
 | [FilterLists](https://filterlists.com/api) | Lists of filters for adblockers and firewalls | No | Yes | Unknown |
-| [HaveIBeenPwned](https://haveibeenpwned.com/API/v2) | Passwords which have previously been exposed in data breaches | No | Yes | Unknown |
+| [HaveIBeenPwned](https://haveibeenpwned.com/API/v3) | Passwords which have previously been exposed in data breaches | No | Yes | Unknown |
```
Comment (Jamim) [inline README.md]: Hello @VorBoto,

As mentioned in #1009, the API is not completely free anymore, so [it requires](https://haveibeenpwned.com/API/v3#Authorisation) providing an [API Key](https://haveibeenpwned.com/API/Key) for some endpoints.

```suggestion
| [HaveIBeenPwned](https://haveibeenpwned.com/API/v3) | Passwords which have previously been exposed in data breaches | `apiKey` | Yes | Unknown |
```

Comment (VorBoto) [reply README.md]: Oh okay, I didn't dive in far enough to catch that.

Comment (yannbertrand) [reply README.md]: @VorBoto can you commit @Jamim suggestion? 🙂

**Comment by VorBoto** [comment]
Sorry I kinda left this just hang.

**Comment by yannbertrand** [comment]
Thank you 😄

**Pull request merged**